src/components/select/utils/with-filter.hoc.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import filterChildren from '../../../utils/filter-children';
+import StyledOption from '../option/option.style';
+import highlightPartOfText from './highlight-part-of-text';
+
+/** Filters wrapped component children based on provided filter text and highlights matching content */
+const withFilter = (WrappedComponent) => {
```

Comment:
What is the benefit of having the `with-filter` hoc, could we re-use this or is it only ever going to be used in `FilterableSelect`?

Reply:
it would be used in the `MultiSelect` and `SelectAsync`. It also allows us to not implement these functionality directly in the SelectList.

Reply:
Great, thanks for explaining.